yarn-project/simulator/src/avm/opcodes/external_calls.ts

```diff
@@ -78,75 +91,32 @@ export class Call extends Instruction {
 
     context.machineState.incrementPc();
   }
-}
-
-export class StaticCall extends Instruction {
-  static type: string = 'STATICCALL';
-  static readonly opcode: Opcode = Opcode.STATICCALL;
-  // Informs (de)serialization. See Instruction.deserialize.
-  static readonly wireFormat: OperandType[] = [
-    OperandType.UINT8,
-    OperandType.UINT8,
-    OperandType.UINT32,
-    OperandType.UINT32,
-    OperandType.UINT32,
-    OperandType.UINT32,
-    OperandType.UINT32,
-    OperandType.UINT32,
-    OperandType.UINT32,
-    /* temporary function selector */
-    OperandType.UINT32,
-  ];
 
-  constructor(
-    private indirect: number,
-    private _gasOffset: number /* Unused due to no formal gas implementation at this moment */,
-    private addrOffset: number,
-    private argsOffset: number,
-    private argsSize: number,
-    private retOffset: number,
-    private retSize: number,
-    private successOffset: number,
-    private temporaryFunctionSelectorOffset: number,
-  ) {
-    super();
+  public get type(): 'CALL' | 'STATICCALL' {
+    const type = super.type;
```

Comment:
Wow, what does super mean here? I thought you'd need sth like `this.(prototype)constructor.type`.

Reply:
That's calling the `type` method from the parent class